### ISSUE TITLE
Invalid memory address or nil pointer dereference when running decrypt.go

### ISSUE THREAD

**Issue by Superb13** [knownunown/tpl-c900-openwrt-install#3]
Getting an error at the start of the extract.sh, when it runs decrypt go using command: go run decrypt.go -in backup.bin -out backup.zz.tmp dec

Used these go versions:
go version go1.10.3 gccgo (Ubuntu 8.3.0-6ubuntu1~18.10.1) 8.3.0 linux/amd64
go version go1.13.4 linux/amd64

Output from go 1.10.3:
go run decrypt.go -in backup.bin -out backup.zz.tmp dec
   panic: runtime error: invalid memory address or nil pointer dereference
   [signal SIGSEGV: segmentation violation code=1 addr=8 pc=140116355550437]

   goroutine 1 [running]:
   panic
         	../../../src/libgo/go/runtime/panic.go:554
   crypto_cipher.NewCBCDecrypter
	       ../../../src/libgo/go/crypto/cipher/cbc.go:101
   main.decrypt
	      /home/superb13/tpl-c900-openwrt-install-master/decrypt.go:66
   main.main
	      /home/superb13/tpl-c900-openwrt-install-master/decrypt.go:49
   exit status 2

Output from 1.13.4:
snap run go run decrypt.go -in backup.bin -out backup.zz.tmp dec
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x49ac26]

goroutine 1 [running]:
crypto/cipher.NewCBCDecrypter(0x0, 0x0, 0x5c9ca0, 0x0, 0x0, 0x50d6a0, 0x5ca760)
	/snap/go/4762/src/crypto/cipher/cbc.go:106 +0x26
main.decrypt(0x5c9ca0, 0x0, 0x0, 0x5c9ca0, 0x0, 0x0, 0xc00009c000, 0x1e00, 0x3e00, 0xc0000a0000, ...)
	/home/superb13/tpl-c900-openwrt-install-master/decrypt.go:66 +0x90
main.main()
	/home/superb13/tpl-c900-openwrt-install-master/decrypt.go:49 +0x59c
exit status 2

**Comment by tnytown** [comment]
Hey, I fixed some error handling -- it should give me some more useful debug info now.
Did you find the key and iv? Did you fill out .encryption_params?

**Comment by Superb13** [comment]
Thanks a lot for taking a look at this! Yes, I was able to find the key and IV. Your additions did help, and I have to say it's something really embarrassing. I scratched my original folder, and started fresh. Every time I ran ./extract.sh, it kept saying something like "no files in <dir>" "backup.bin.tmp not found". I ran sudo, and same thing. Then I did a chmod 755 on the whole folder, and that worked! It created the XML file, and didn't give me any issues with packaging the config file. Odd, since I was the owner of all of the files in the folder. Going to upload it to the router, and see what I've got. Thanks again for your time!
